### PR TITLE
Internal: AI builder prompt autocomplete [ED-11546]

### PR DESCRIPTION
**Pull request by rotemee**

### CODE REVIEW

**Review by StyleShit — APPROVED** [review]

**Review by StyleShit — CHANGES_REQUESTED** [review]
.

File: modules/ai/assets/js/editor/pages/form-layout/index.js [hunk 1]
```diff
@@ -38,6 +42,10 @@ const FormLayout = ( { onClose, onInsert, onGenerationStart, onGenerationEnd, on
 	const handleSubmit = ( event ) => {
 		event.preventDefault();
 
+		if ( '' === prompt ) {
```
[the remark this answers is not in the document]
Comment (rotemee) [reply modules/ai/assets/js/editor/pages/form-layout/index.js:45]: Done.

File: modules/ai/module.php [hunk 2]
```diff
@@ -655,9 +655,15 @@ public function ajax_ai_generate_layout( $data ) {
 			throw new \Exception( $result->get_error_message() );
 		}
 
+		$templates_data = array_map( function( $template_data ) {
```
[the remark this answers is not in the document]
Comment (rotemee) [reply modules/ai/module.php:658]: Done.

File: modules/ai/assets/js/editor/pages/form-layout/components/prompt-autocomplete.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			freeSolo
+			fullWidth
+			disableClearable
+			open={ showSuggestions }
+			onClose={ () => setShowSuggestions( false ) }
+			onKeyDown={ ( e ) => {
+				if ( 'Enter' === e.key && ! e.shiftKey && ! showSuggestions ) {
+					onSubmit( e );
+				} else if ( '/' === e.key && '' === e.target.value ) {
+					e.preventDefault();
```
[the remark this answers is not in the document]
Comment (rotemee) [reply modules/ai/assets/js/editor/pages/form-layout/components/prompt-autocomplete.js:46]: For now, I will not add the slash to the prompt.

**Review by StyleShit — APPROVED** [review]

**Pull request merged**